install_script.sh.template

```diff
@@ -63,6 +63,68 @@ function report(){
   fi
 }
 
+function report_telemetry() {
+  if [ "$DD_INSTRUMENTATION_TELEMETRY_ENABLED" == "false" ] || \
+    [ "$site" == "ddog-gov.com" ] || \
+    [ -z "${apikey}" ] || \
+    [ -z "$telemetry_url" ]; then
+    return
+  fi
+
+  if [ -n "$agent_minor_version" ] ; then
+    safe_agent_version=$(echo -n "$agent_major_version.$agent_minor_version" | tr '\n' ' ' | tr '"' '_')
+  else
+    safe_agent_version=$(echo -n "$agent_major_version" | tr '\n' ' ' | tr '"' '_')
+  fi
+
+  if [ -z "${ERROR_CODE}" ] ; then
+    telemetry_event="
+{
+   \"request_type\": \"apm-onboarding-event\",
+   \"api_version\": \"v1\",
+   \"payload\": {
+       \"event_name\": \"agent.installation.success\",
+       \"tags\": {
+           \"agent_platform\": \"native\",
+           \"agent_version\": \"$safe_agent_version\",
+           \"script_version\": \"$install_script_version\"
+       }
+   }
+}
+"
+  else
+    safe_error_message=$(echo -n "$ERROR_MESSAGE" | tr '\n' ' ' | tr '"' '_')
+    telemetry_event="
+{
+   \"request_type\": \"apm-onboarding-event\",
+   \"api_version\": \"v1\",
+   \"payload\": {
+       \"event_name\": \"agent.installation.error\",
+       \"tags\": {
+           \"agent_platform\": \"native\",
+           \"agent_version\": \"$safe_agent_version\",
+           \"script_version\": \"$install_script_version\"
+       },
+       \"error\": {
+          \"code\": $ERROR_CODE,
+          \"message\": \"$safe_error_message\"
+       }
+   }
+}
+"
+  fi
+
+   (cat <<END
+       $telemetry_event
+END
+       ) | curl --fail --silent --output /dev/null \
+    "$telemetry_url" \
+    --header 'Content-Type: application/json' \
+    --header "DD-Api-Key: $apikey" \
```

Comment:
How important is it for you that the `$apikey` here is the same as the one used by the Agent? 

The main reason I'm asking this is that the `DD_API_KEY` environment variable passed to the script (which becomes this `$apikey` variable) may not be the one that is set in the `datadog.yaml` file.

For instance, if a `datadog.yaml` config file already exists (due to a prior install), then the `DD_API_KEY` parameter of the script is ignored, and no changes to the `datadog.yaml` file are made, see this code snippet: https://github.com/DataDog/agent-linux-install-script/blob/0a6f442b2182f9a70cd8ed3ea2f50609bb3677c7/install_script.sh.template#L825-L842

In fact, we don't require the `DD_API_KEY` variable at all for upgrades, therefore the `$apikey` may be empty: https://github.com/DataDog/agent-linux-install-script/blob/0a6f442b2182f9a70cd8ed3ea2f50609bb3677c7/install_script.sh.template#L403-L409

Reply:
The api key is used to determine the org for the telemetry. If the api keys are different, I think in almost all cases, they'll belong to the same org.

In the extremely rare case they do belong to different orgs, I think we can live with that. The telemetry will be mostly used in aggregate across all orgs.

For upgrades, I think partial data is okay. I asked the rest of the team if we should just block telemetry on upgrades altogether instead of partial data.

Reply:
The team agrees that we shouldn't worry about the update case